spectree/plugins/flask_plugin.py

```diff
@@ -111,8 +111,18 @@ def parse_path(self, route):
         return ''.join(subs), parameters
 
     def request_validation(self, request, query, json, headers, cookies):
+        from werkzeug.exceptions import BadRequest
+
         req_query = request.args or {}
-        req_json = request.get_json() or {}
+
+        try:
+            # if content-type is set to application/json but json is passed in below crashes
+            # it is technically intentional by flask, however this can have unexpected results for clients 
+            # that already set these properties incorrectly
+            req_json = request.get_json() or {}
+        except BadRequest:
+            req_json = {}
+
```

Comment:
Maybe can use `request.get_json(silent=True)`? So it will ignore the parsing errors and return `None`

Reply:
yes I like that better